### PR TITLE
Pdct 811/matricising the current ci process

### PR DESCRIPTION
**Pull request by katybaulch**
- Matricised build & test workflow for each theme
- Updated frontend dockerfile to run as non root user & changed working dir to default
- Simplified how cypress e2e tests are run in docker
- Updated tagging to push each theme image to their own ECR repo

[PDCT-807 Update CI to build all themes](https://linear.app/climate-policy-radar/issue/PDCT-807/update-ci-to-build-all-themes)

[PDCT-810 Investigate semver tagging for themeing](https://linear.app/climate-policy-radar/issue/PDCT-810/investigate-semver-tagging-for-themeing)

[PDCT-811 Matricising the current CI process](https://linear.app/climate-policy-radar/issue/PDCT-811/matricising-the-current-ci-process)

[PDCT-812 Run E2E tests for each theme](https://linear.app/climate-policy-radar/issue/PDCT-812/run-e2e-tests-for-each-theme)

To do at a later date, as it doesn't look so straightforward: https://linear.app/climate-policy-radar/issue/PDCT-824/read-themes-from-file-into-github-ci-matrix

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -69,6 +67,6 @@ jobs:
 
       - name: Push Images to ECR
         run: |
-          .github/retag-and-push.sh navigator-frontend latest
+          .github/retag-and-push.sh navigator-frontend-${{ matrix.theme }} latest
```
Comment (olaughter) [inline .github/workflows/ci.yml:70]: I believe this will require manually setting up an ecr repo for each new theme, just an fyi

Comment (katybaulch) [reply .github/workflows/ci.yml:70]: Have done this, thanks!!

File: e2e/cypress/e2e/redirects/redirects.cy.js [hunk 2]
```diff
@@ -4,6 +4,9 @@
 const startingIndex = 0;
 const createTestCases = false;
 
+// TODO: write a better test script for redirects that doesn't run through the entire set of redirects
+// For example: we could use a sample of redirects to test against
```
Comment (diversemix) [inline e2e/cypress/e2e/redirects/redirects.cy.js:8]: Do we have a ticket for this?

**Review by diversemix — APPROVED** [review]
lgtm :+1:

File: e2e/cypress/e2e/redirects/redirects.cy.js (hunk 2 above)
Comment (PatFawbertMills) [reply e2e/cypress/e2e/redirects/redirects.cy.js:8]: We do now 😁

**Pull request merged**